lib/travis/build/script/elm.rb

```diff
@@ -0,0 +1,183 @@
+          end
+
+          def elm_major_version
+            elm_version_number 0
+          end
+
+          def elm_minor_version
+            elm_version_number 1
+          end
+
+          def elm_patch_version
+            elm_version_number 2
+          end
+
+          def elm_test_version
+            config[:elm_test].to_s
+          end
+
+          def elm_format_version
+            config[:elm_format].to_s
+          end
+
+          def install_elm
+            npm_install "-g elm@elm-#{elm_version}"
+
+            convert_binary_to_sysconfcpus 'elm'
+
+            # Beginning with Elm 0.19, there's one `elm` binary and that's it.
+            # In that case, there won't be any files to convert here!
+            if elm_major_version == 0 && elm_minor_version <= 18
+              convert_binary_to_sysconfcpus 'elm-make'
+              convert_binary_to_sysconfcpus 'elm-package'
+              convert_binary_to_sysconfcpus 'elm-format'
+            end
+          end
+
+          def install_elm_format
+            npm_install "-g elm-format@elm-#{elm_format_version}"
+
+            convert_binary_to_sysconfcpus 'elm-format'
+          end
+
+          def install_elm_test
+              sh.if "$(vers2int $(echo `node --version` | tr -d 'v')) -lt $(vers2int #{ELM_TEST_REQUIRED_NODE_VERSION})" do
+                sh.echo "Node.js version $(node --version) does not meet requirement for elm-test." \
+                  " Please use Node.js #{ELM_TEST_REQUIRED_NODE_VERSION} or later.", ansi: :red
+              end
+              sh.else do
+                sh.if "-z \"$(command -v elm-test)\"" do
+                  npm_install "-g elm-test@#{elm_test_version}"
+
+                  convert_binary_to_sysconfcpus 'elm-test'
+                end
+              end
+          end
+
+          def convert_binary_to_sysconfcpus(binary_name)
+            # Wrap the binary in a call to sysconfcpus -n 2
+            # to work around https://github.com/travis-ci/travis-ci/issues/6656
+            sh.mv "$(npm config get prefix)/bin/#{binary_name}", "$(npm config get prefix)/bin/#{binary_name}-old"
+
+            # Use printf instead of echo here.
+            # see https://github.com/rtfeldman/node-elm-compiler/pull/50
+            sh.cmd ('printf "#\041/bin/bash\n\necho \"Running ' + binary_name + ' with sysconfcpus -n 2\"\n\n$TRAVIS_BUILD_DIR/sysconfcpus/bin/sysconfcpus -n 2 ' + binary_name + '-old \"\$@\"" > $(npm config get prefix)/bin/' + binary_name)
+            sh.chmod '+x', "$(npm config get prefix)/bin/#{binary_name}"
+          end
+
+          def install_sysconfcpus
+            sh.fold 'sysconfcpus' do
+              # this is a prerequisite for the convert_binary_to_sysconfcpus method
+              # which provides an epic build time improvement - see https://github.com/elm-lang/elm-compiler/issues/1473#issuecomment-245704142
+              sh.cmd 'git clone https://github.com/obmarg/libsysconfcpus.git', retry: true
```

Comment:
Ditto for this fold.

Reply:
Added in e386ff219defbe58bbe931cc0b66b57e866d818c